File: ClientApp/src/app/clases/horasdia.ts
export class horasdia
{
    id: number;
    hora: string; 
    numorden: number;  

    constructor(obj:any)
    {
        this.id= +obj.id;
        this.hora =  obj.hora;
        this.numorden = +obj.numorden;   
    }
}

File: modelo/horasdia.cs
using System.Data;
using System;
public class horasdia : oObjeto
{
 
    public string hora { get; set; } 
    public int numorden {get;set;} 

    public horasdia() { }
    public horasdia(DataRow dr)
    {
        this.Id = Convert.ToInt32(dr["id"]);
        this.hora = dr["hora"].ToString();
        this.numorden=Convert.ToInt32(dr["numorden"]); 
    }

}